packet/loadbalancers.go

```diff
@@ -100,46 +100,58 @@ func (l *loadBalancers) serviceReconciler() serviceReconciler {
 // by adding it to or removing it from the known metallb configmap
 func (l *loadBalancers) reconcileNodes(nodes []*v1.Node, remove bool) error {
 	var (
-		peers []string
-		err   error
+		peers        []string
+		err          error
+		changedNodes bool
+		changed      bool
```

Comment:
I could be overlooking something, but it appears that changedNodes is only every set or used when changed is set, and in all cases it matches the value of changed.

Reply:
You did miss. The `changed` is set inside a loop for each node. `changedNodes` is the overall "have any nodes changed". In each loop (this is from the add):

```go
			config, changed = addNodePeer(config, node.Name, l.localASN, l.peerASN, peers...)
			if !changed {
				klog.V(2).Infof("loadbalancers.reconcileNodes(): config unchanged for add %s", node.Name)
			} else {
				klog.V(2).Infof("loadbalancers.reconcileNodes(): config changed for add %s", node.Name)
				changedNodes = true
			}
```

So `changed` captures "did this node change", which `changedNodes` is set to `true` if at least one of them had `changed`. We then use `changedNodes` to see if _any_ if the nodes have changed (thus requiring us to update the configmap).